src/features/virtualDocumentTracker.ts

```diff
@@ -0,0 +1,91 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import {workspace, TextDocument} from 'vscode';
+import {OmniSharpServer} from '../omnisharp/server';
+import * as serverUtils from '../omnisharp/utils';
+import { FileChangeType } from '../omnisharp/protocol';
+import { IDisposable } from '../Disposable';
+import CompositeDisposable from '../CompositeDisposable';
+
+function trackCurrentVirtualDocuments(server: OmniSharpServer) {
+    let registration = server.onProjectAdded(() => {
```

Comment:
This is going to add the existing virtual documents every time O# discovers a project, which can happen multiple times. There is no "done loading" event today, unfortunately.

Reply:
(We should test this in a multi-project + razor scenario and verify o# doesn't do anything weird if you add a buffer twice)

Reply:
This code is only ever called once for the first project. See at the end of the method where it disposes its own registration to the method.

Reply:
Ah, right. Thanks.